lib/commands/new/command.js

```diff
@@ -1,47 +1,226 @@
+    await this.displayCompletionMessage(projectName, features, projectFolder);
+  }
+
+  async ensureProjectName(name) {
+    name = (name || '').trim();
+    if (name) return name;
+    if (this.unattended) return 'aurelia-app';
+
+    return await this.ui.ensureAnswer(
+      name,
+      'Please enter a name for your new project:',
+      'aurelia-app'
     );
+  }
+
+  async confirmToWriteOnExistingFolder(projectName) {
+    let question;
+    let yesHint;
+
+    if (!this.here && fs.isFile(projectName)) {
+      throw new Error(`There is an existing file named as ${projectName}, cannot create folder ${projectName}.`);
+    } else if (!this.here && fs.isDirectory(projectName)) {
+      if (this.unattended) {
+        throw new Error(`The '${projectName}' folder already exists. Please make sure it's not there for unattended mode.`);
+      }
+      question = `WARNING: The '${projectName}' folder already exists. Would you like to create the project in this folder?`;
+      yesHint = `Continue even though the '${projectName}' directory already exists.`;
+    } else if (this.here && fs.readdirSync(process.cwd()).length > 0) {
+      // Default answer is yes in unattended mode
+      if (this.unattended) return true;
+      question = 'WARNING: The current directory is not empty. Would you like to create the project in this folder?';
+      yesHint = 'Continue even though the current directory is not empty.';
+    }
+
+    if (!question) return true;
+
+    const confirm = await this.ui.question(
+      c.red(question),
+      [
+        {value: 'abort', message: 'Abort', hint: 'Aborts the new project wizard.'},
+        {value: 'continue', message: 'Continue', hint: c.red(yesHint)}
+      ]
+    );
+
+    return confirm === 'continue';
+  }
 
-    let state = {};
+  async showSummary(projectName, features) {
+    // Ensure to wrap projectName with quotes if there is white space in it.
+    const wrappedName = projectName.match(/\s/) ? JSON.stringify(projectName) : projectName;
+    const feats = await minFeatures(features, this.plugin);
 
-    if (this.options.hasFlag('here')) {
-      state.name = this.options.originalBaseDir;
-      state.defaultOrCustomOverride = 'custom';
+    await this.ui.log('');
+    await this.logTitle('📖', 'Project Summary');
+    await this.logBody(`This project was bootstrapped based on following selected features:
+${c.green(features.join(' '))}
+Next time, you can generate a new project faster in unattended mode with the same features:
+${cmd(`au new ${wrappedName} --unattended${feats.length ? ' --select ' : ''}${feats.join(',')}${this.plugin ? ' --plugin' : ''}${this.here ? ' --here' : ''}`)}
+Or use the shorter form:
+${cmd(`au new ${wrappedName} -u${feats.length ? ' -s ' : ''}${feats.join(',')}${this.plugin ? ' -p' : ''}${this.here ? ' -h' : ''}`)}
+Note the command might not supply all the selected features, as some of them could be from default offering.`);
```

Comment:
Not sure if I understand what this means. Shouldn't you always get the features that are specified in the command?

Reply:
If you don't specify some features, the default choice will be picked automatically in unattended mode.

So `au new some --unattended` will generate app with all default choices.
`au new some --unattended --select jest,vscode` will generate an app as if you chose "Default ESNEXT app".
`au new some --unattended --select typescript,jest,vscode` will generate an app as if you chose "Default Typescript app".

Reply:
Ah ok :) I didn't understand that at first from reading it, but it could very well be just me

Reply:
Maybe something like: Some features get installed by default, so those are omitted from the command,

Reply:
Updated accordingly.